lib/helper/WebDriverIO.js

```diff
@@ -1090,12 +1224,56 @@ class WebDriverIO extends Helper {
    * ```
    */
   dragAndDrop(srcElement, destElement) {
-    return this.browser.dragAndDrop(
-      withStrictLocator(srcElement),
-      withStrictLocator(destElement)
-    );
+    let client = this.browser;
+    let this2 = this;
+
+
+    if (client.isMobile) {
```

Comment:
It's belong to this PR. This refactoring gives opportunity to use dragAndDrop in `within` correctly